### PR TITLE
Fix: Ser/Deser of the UserFeedback from cached envelope

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: Ser/Deser of the UserFeedback from cached envelope #1611

## :bulb: Motivation and Context
Fix https://github.com/getsentry/sentry-dart/issues/534

## :pencil: Checklist

- [X] I reviewed the submitted code
- [X] I added tests to verify the changes
- [ ] I updated the docs if needed
- [X] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@ueman

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Fix: set min sdk version of sentry-android-fragment to API 14 (#1608)
+* Fix: Ser/Deser of the UserFeedback from cached envelope
```
Comment (marandaneto) [inline CHANGELOG.md:6]:
```suggestion
* Fix: Ser/Deser of the UserFeedback from cached envelope (#1611)
```

File: sentry/src/main/java/io/sentry/SentryEnvelopeItemHeaderAdapter.java [hunk 2]
```diff
@@ -70,9 +70,21 @@ public void write(JsonWriter writer, SentryEnvelopeItemHeader value) throws IOEx
           break;
         case "type":
           try {
-            final String nextString = StringUtils.capitalize(reader.nextString());
+            String nextString = reader.nextString();
+
             if (nextString != null) {
-              type = SentryItemType.valueOf(nextString);
+              // special case the String 'user_report since the enum is called UserFeedback
+              // instead of UserReport, this is gonna be fixed when we move away from reflection.
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/SentryEnvelopeItemHeaderAdapter.java:77]: https://github.com/getsentry/sentry-java/pull/1554

File: sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt [hunk 3]
```diff
@@ -137,4 +135,25 @@ abcdefghij""".toInputStream()
         assertEquals(10, secondItem.header.length)
         assertEquals(10, secondItem.data.size)
     }
+
+    @Test
+    fun `deserializes an user feedback`() {
+        val envelope = readEnvelope("envelope-feedback.txt")
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt:141]: `l`: I think we could rename the `envelope-feedback.txt` to `envelope-feedback-from-dart.json` to reflect it is an envelope from dart and it contains json.

File: sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt [hunk 4]
```diff
@@ -137,4 +135,25 @@ abcdefghij""".toInputStream()
         assertEquals(10, secondItem.header.length)
         assertEquals(10, secondItem.data.size)
     }
+
+    @Test
+    fun `deserializes an user feedback`() {
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt:140]: `l`: I think the test name could reflect the envelope is coming from dart.

File: sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt [hunk 5]
```diff
@@ -137,4 +135,25 @@ abcdefghij""".toInputStream()
...
+        assertEquals("bdd63725a2b84c1eabd761106e17d390", envelope.header.eventId.toString())
+        assertEquals(1, envelope.items.count())
+        val firstItem = envelope.items.elementAt(0)
+        assertEquals(SentryItemType.UserFeedback, firstItem.header.type)
+        assertEquals("application/json", firstItem.header.contentType)
+        assertEquals(103, firstItem.header.length)
+        assertEquals(103, firstItem.data.size)
+    }
+
+    private fun readEnvelope(fileName: String): SentryEnvelope? {
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt:153]: Nice 👍

**Review by philipphofmann — APPROVED** [review]
LGTM, just minor naming suggestions.

File: sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt (hunk 3 above)
Comment (marandaneto) [reply sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt:141]: the same issue would happen with an envelope from react-native, or sentry-native, so it's rather a valid feedback envelope, I'd keep as it is

File: sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt (hunk 4 above)
Comment (marandaneto) [reply sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt:140]: same as https://github.com/getsentry/sentry-java/pull/1611#discussion_r672971707

File: sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt (hunk 3 above)
Comment (philipphofmann) [reply sentry/src/test/java/io/sentry/SentryEnvelopeTest.kt:141]: Fine with me. 🚢

**Pull request merged**